drivers/flash/soc_flash_nrf.c

```diff
@@ -327,11 +330,21 @@ static void time_slot_callback_helper(u32_t ticks_at_expire, u32_t remainder,
 
 static int work_in_time_slice(struct flash_op_desc *p_flash_op_desc)
 {
+	u32_t flash_interval;
 	u8_t instance_index;
+	u32_t flash_slot;
 	u8_t ticker_id;
 	int result;
 	u32_t err;
 
+	if (p_flash_op_desc->handler == erase_op) {
+		flash_interval = FLASH_INTERVAL_ERASE;
```

Comment:
what about extend `struct flash_op_desc` by `interval` and `slot` and transferring this values from the caller?

Reply:
These value is build time constants. For erase is constant, and for write its the BT lowest connection interval of 7.5ms. But for better code, and as long as the parameter is in stored and accessed from code space, we can have it in the struct. 

@joerchan can you handle the request and push to my branch (append your signed-off-by)